File: Objective-C/TCMPPDemo/Utils/DemoUtils.m
//
//  DemoUtils.m
//  TCMPPDemo
//
//  Created by stonelshi on 2023/12/6.
//

#import "DemoUtils.h"

@implementation DemoUtils


+(NSString *)convertToJsonData:(NSDictionary *)dict
{
    NSError *error;
    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dict options:NSJSONWritingPrettyPrinted error:&error];
    NSString *jsonString;
    if (!jsonData) {
        NSLog(@"%@",error);
    }else{
        jsonString = [[NSString alloc]initWithData:jsonData encoding:NSUTF8StringEncoding];
    }

    NSMutableString *mutStr = [NSMutableString stringWithString:jsonString];
    NSRange range = {0,jsonString.length};
    
    [mutStr replaceOccurrencesOfString:@" " withString:@"" options:NSLiteralSearch range:range];
    NSRange range2 = {0,mutStr.length};
    [mutStr replaceOccurrencesOfString:@"\n" withString:@"" options:NSLiteralSearch range:range2];
    
    NSRange range3 = {0,mutStr.length};
    [mutStr replaceOccurrencesOfString:@"\\/\\/" withString:@"//" options:NSLiteralSearch range:range3];
    
    return mutStr;
}


+ (void)executeOnMainThread:(dispatch_block_t)block {
    if ([NSThread isMainThread]) {
            block();
        } else {
            dispatch_async(dispatch_get_main_queue(), block);
        }
}


+(NSString *)validNSString:(NSString *)str {
    return str ?: @"";
}

@end

File: Swift/TCMPPDemo-Swift/Class/Utils/DemoUtils.swift
//
//  DemoUtils.swift
//  TCMPPDemo-Swift
//
//  Created by gavinjwxu on 2024/8/13.
//

import Foundation

class DemoUtils {
    static func convertToJsonData(_ dict: [String: Any]) -> String {
        var jsonString = ""
        if let jsonData = try? JSONSerialization.data(withJSONObject: dict, options: .prettyPrinted) {
            jsonString = String(data: jsonData, encoding: .utf8) ?? ""
        } else {
            print("Error serializing JSON")
        }

        let mutStr = NSMutableString(string: jsonString)
        let fullRange = NSRange(location: 0, length: mutStr.length)
        mutStr.replaceOccurrences(of: " ", with: "", options: .literal, range: fullRange)
        mutStr.replaceOccurrences(of: "\n", with: "", options: .literal, range: fullRange)
        mutStr.replaceOccurrences(of: "\\/\\/", with: "//", options: .literal, range: fullRange)

        return mutStr as String
    }

    static func executeOnMainThread(_ block: @escaping () -> Void) {
        if Thread.isMainThread {
            block()
        } else {
            DispatchQueue.main.async(execute: block)
        }
    }

    static func validNSString(_ str: String?) -> String {
        return str ?? ""
    }
}
